Static/TableViewController.swift

```diff
@@ -1,19 +1,96 @@
 import UIKit
 
 /// Table view controller with a `DataSource` setup to use its `tableView`.
-public class TableViewController: UITableViewController {
+public class TableViewController: UIViewController, UITableViewDataSource, UITableViewDelegate {
 
     // MARK: - Properties
 
+    /// Returns the table view managed by the controller object.
+    public let tableView: UITableView
+
+    /// A Boolean value indicating if the controller clears the selection when the table appears.
+    ///
+    /// The default value of this property is YES. When YES, the table view controller clears the table’s current selection when it receives a viewWillAppear: message. Setting this property to NO preserves the selection.
+    public var clearsSelectionOnViewWillAppear: Bool = true
+
     /// Table view data source.
     public let dataSource = DataSource()
 
 
+    // MARK: - Initialization
+
+    public init(style: UITableViewStyle) {
+        tableView = UITableView(frame: .zeroRect, style: style)
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    public override init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: NSBundle?) {
+        tableView = UITableView(frame: .zeroRect, style: .Plain)
+        super.init(nibName: nibNameOrNil, bundle: nibBundleOrNil)
+    }
+
+    public required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    public convenience init() {
+        self.init(style: .Plain)
+    }
+
+
     // MARK: - UIViewController
 
+    public override func loadView() {
+        tableView.autoresizingMask = [.FlexibleWidth, .FlexibleHeight]
+        tableView.delegate = self
+        tableView.dataSource = self
+        view = tableView
+    }
+
     public override func viewDidLoad() {
         super.viewDidLoad()
-
         dataSource.tableView = tableView
     }
+
+    public override func viewWillAppear(animated: Bool) {
+        super.viewWillAppear(animated)
+        performInitialLoad()
```

Comment:
this feels hacky with a reason. Why are we doing it in `viewWillAppear ` in a `onceToken`